src/platforms/rp2040/src/lib/networkdriver.c

```diff
@@ -419,6 +479,52 @@ static term start_ap(term ap_config, GlobalContext *global)
         }
     }
 
+    if (term_is_invalid_term(hostname_term)) {
+        driver_data->ap_hostname = malloc(DEFAULT_HOSTNAME_SIZE);
+        if (IS_NULL_PTR(driver_data->ap_hostname)) {
+            free(ssid);
+            free(psk);
+            return OUT_OF_MEMORY_ATOM;
+        } else {
+            term error = get_default_device_name(driver_data->ap_hostname, global);
+            if (error != OK_ATOM) {
+                free(ssid);
+                free(psk);
+                free(driver_data->ap_hostname);
+                return error;
+            }
+        }
+    } else {
+        int ok = 0;
+        char *buf = malloc(MAX_HOSTNAME_SIZE);
+        if (IS_NULL_PTR(buf)) {
+            free(ssid);
+            free(psk);
+            return OUT_OF_MEMORY_ATOM;
+        } else {
+            buf = interop_term_to_string(hostname_term, &ok);
+        }
+        if (UNLIKELY(!ok || buf == NULL)) {
+            free(ssid);
+            free(psk);
+            if (buf != NULL) {
+                free(buf);
+            }
+            return BADARG_ATOM;
+        }
+        driver_data->ap_hostname = malloc(strlen(buf) + 1);
```

Comment:
You could use strdup(3) which would do the computation of the length, the allocation and the copy in a single call. Alternatively, if it's not available in some SDKs (it's part of C89), you could save the length to avoid computing it twice.